### PR TITLE
docs: improve accessibility of animations example

### PR DESCRIPTION
**Pull request by kapunahelewong**
- Changes labels to explicitly state intent.
- Changes markup to make it more semantic and consistent with other examples.
- Updates CSS and removes duplicate/unused styles.
- Improves contrast.
- Adds `type="text"` so that boilerplate styles are inherited.
- Removes images that are outdated and that rely on too much text. The copy in the doc explains the concept clearly.

## Before (contrast and missing/unreferenced label issues) 
Screen Shot 2021-03-30 at 5 08 16 PM
Screen Shot 2021-03-30 at 5 07 52 PM
Screen Shot 2021-03-30 at 5 07 25 PM

## After

Screen Shot 2021-03-30 at 3 19 11 PM
Screen Shot 2021-03-30 at 3 24 37 PM
Screen Shot 2021-03-30 at 3 24 46 PM
Screen Shot 2021-03-30 at 4 17 21 PM

### CODE REVIEW

**Review by aikithoughts — APPROVED** [review]
Reviewed-for: global-docs-approvers

File: aio/content/examples/animations/src/app/app.component.css [hunk 1]
```diff
@@ -1,7 +1,3 @@
-:host {
-  display: block;
-}
-
-section {
-  margin-top: 100px;
+nav a {
+  padding: .7rem;
```
Comment (gkalpak) [inline aio/content/examples/animations/src/app/app.component.css:2]: I couldn't help noticing that each example seems to use a different padding for `nav a` items, so now I am wondering why that is 😇

File: aio/content/examples/animations/src/app/app.component.html [hunk 2]
```diff
@@ -1,6 +1,10 @@
 <h1>Animations</h1>
 
-Toggle All Animations <input type="checkbox" [checked]="!animationsDisabled" (click)="toggleAnimations()"/>
+<input type="checkbox"
+        id="animation-toggle"
```
Comment (gkalpak) [inline aio/content/examples/animations/src/app/app.component.html:4]:
```suggestion
<input type="checkbox"
       id="animation-toggle"
```

File: aio/content/examples/animations/src/app/hero-list-page.component.css [hunk 3]
```diff
@@ -1,51 +1,33 @@
...
-}
-
 .heroes .badge {
   display: inline-block;
   font-size: small;
   color: white;
   padding: 0.8em 0.7em 0 0.7em;
-  background-color: #607D8B;
-  line-height: 1em;
+  background-color: #3D5157;
```
Comment (gkalpak) [inline aio/content/examples/animations/src/app/hero-list-page.component.css:30]: Super-nit (to be consistent with other examples): 

```suggestion
  background-color: #3d5157;
```

File: aio/content/examples/animations/src/app/hero-list-page.component.html [hunk 4]
```diff
@@ -2,18 +2,21 @@
 <h2>Filter/Stagger</h2>
 
 <form>
-  <input #criteria (input)="updateCriteria(criteria.value)" placeholder="Search Heroes" />
-</form>    
+  <label for="search">Search heroes: </label>
+  <input type="text"
+         #criteria (input)="updateCriteria(criteria.value)"       placeholder="Search heroes"
```
Comment (gkalpak) [inline aio/content/examples/animations/src/app/hero-list-page.component.html:7]: You probably meant to split these into three separate lines :grin:

File: aio/content/examples/animations/src/app/hero-list-page.component.html [hunk 5]
```diff
@@ -2,18 +2,21 @@
...
+         id="search">
```
Comment (gkalpak) [inline aio/content/examples/animations/src/app/hero-list-page.component.html:8]: Nit: I think putting this right after the `type` (on a new line) would be more consistent with other similar occurrences.

File: aio/content/examples/animations/src/app/open-close-page.component.ts [hunk 6]
```diff
@@ -5,7 +5,8 @@ import { Component } from '@angular/core';
   template: `
     <section>
       <h2>Open Close Component</h2>
-      <input type="checkbox" [checked]="logging" (click)="toggleLogging()"/> Console Log Animation Events
+      <input id="log-checkbox" type="checkbox" [checked]="logging" (click)="toggleLogging()"/>
```
Comment (gkalpak) [inline aio/content/examples/animations/src/app/open-close-page.component.ts:8]: Not super-important, but I think it would be nice (and beneficial :wink:) to use a somewhat consistent attribute ordering (for example, on `inputs` put `type` first, then `id` (if present), etc.).

File: aio/content/examples/animations/src/app/open-close.component.css [hunk 7]
```diff
@@ -1,5 +1,6 @@
-:host {
+button {
```
Comment (gkalpak) [inline aio/content/examples/animations/src/app/open-close.component.css:1]: OOC why this change? It seems strange to add a top margin to all buttons generically.

AFAICT, the intention is to add a margin on the `app-open-close` element and the fact that it only contains one button and it happens to be at the top is an "implementation detail" (as far as styles are concerned).

So, why not add the margin in the host element?

File: aio/content/guide/animations.md [hunk 8]
```diff
@@ -63,17 +63,13 @@ In the component file, add a metadata property called `animations:` within the `
 <code-example path="animations/src/app/app.component.ts" header="src/app/app.component.ts" region="decorator" language="typescript">
 </code-example>
 
-## Animating a simple transition
+## Animating a transition
 
-Let's animate a simple transition that changes a single HTML element from one state to another. For example, you can specify that a button displays either **Open** or **Closed** based on the user's last action. When the button is in the `open` state, it's visible and yellow. When it's the `closed` state, it's transparent and green.
+Let's animate a transition that changes a single HTML element from one state to another. For example, you can specify that a button displays either **Open** or **Closed** based on the user's last action. When the button is in the `open` state, it's visible and yellow. When it's the `closed` state, it's transparent and blue.
```
Comment (gkalpak) [inline aio/content/guide/animations.md:68]: Nit: I think "translucent and blue" would be more accurate 😇

**Review by gkalpak — CHANGES_REQUESTED** [review]
🦄

File: aio/content/examples/animations/src/app/app.component.html [hunk 9]
```diff
@@ -1,6 +1,10 @@
...
+       [checked]="!animationsDisabled"
+       (click)="toggleAnimations()">
+<label for="animation-toggle">Toggle All Animations</label>
```
Comment (jelbourn) [inline aio/content/examples/animations/src/app/app.component.html:7]: optional: you can also associate an input with a label by putting the input inside the label element, e.g.
```html
<label>
  <input ...> Toggle all animations
</label>
```

This can keep you from needing to add `id` and `for` attributes

**Review by jelbourn — APPROVED** [review]
overall lgtm aside from @gkalpak's comments

File: aio/content/examples/animations/src/app/app.component.css (hunk 1 above)
Comment (kapunahelewong) [reply aio/content/examples/animations/src/app/app.component.css:2]: The boilerplate padding is a bit skinny and dated, but I didn't change it at that level out of caution in case it would break one of the examples. My thought on this particular change was to to update the boilerplate once I'd gone through the examples and had a feel for how many of these there are and how they might be effected. Would you prefer they be left with the boilerplate padding instead? I can remove this one, though my preference is the first way. Your call, @gkalpak. 🙂

File: aio/content/examples/animations/src/app/open-close.component.css (hunk 7 above)
Comment (kapunahelewong) [reply aio/content/examples/animations/src/app/open-close.component.css:1]: Imho, being on the `button` is more obvious, especially to the developer has only a cursory understanding of CSS, but for you, @gkalpak, I'll put it back. Just for you though! 😁

**Comment by Toledochrist** [comment]
Thanks

File: aio/content/examples/animations/src/app/app.component.html (hunk 9 above)
Comment (kapunahelewong) [reply aio/content/examples/animations/src/app/app.component.html:7]: Agreed, though my understanding is that some screen readers stumble on this. 

Though Jeremy already knows this, for context for anyone wondering, check out the links from this comment for reference: https://github.com/angular/angular/pull/41252#discussion_r605723196
